### ISSUE TITLE
Chrome seems to have a hard limit when taking screenshots of long pages

### ISSUE THREAD

**Issue by guilhermehn** [puppeteer/puppeteer#359]
I'm using puppeteer (previously was using pure CDP) to take screenshots from all the endpoints of my application and some pages that contain legal info and EULA are really tall. Like really tall:

![image](https://user-images.githubusercontent.com/685012/29429858-7f967c66-8368-11e7-82e9-ae412cd48c0e.png)

And the resulting screenshots are only 16384px high. When debugging the headless instance with devTools the page seems to be fully rendered.

Here's the minimal snippet:

``` js
const browser = await puppeteer.launch();
const page = await browser.newPage();
await page.goto(url, {
	waitUntil: 'load'
});
await page.setViewport({
	width: size, // 320, 768, 1024, 1280
	height: 600
});
const screenshot = await page.screenshot({
	fullPage: true
});
browser.close();
fs.writeFileSync('./results/screenshot.png', screenshot);
```

Some of the tall pages screenshots result in a completely white screenshot. If I set the viewport using the `page.evaluate` method, it seems to work, but the height limit of 16384px still holds true.

``` js
const browser = await puppeteer.launch();
const page = await browser.newPage();
await page.goto(url, {
	waitUntil: 'load'
});
await page.setViewport({
	width: size, // 320, 768, 1024, 1280
	height: 600
});
await page.setViewport({
	width: size,
	height: await page.evaluate(() => document.body.clientHeight)
});
const screenshot = await page.screenshot();
browser.close();
fs.writeFileSync('./results/screenshot.png', screenshot);
```

But regardless of the technique used, the screenshots' height are all capped to 16384px.

**Comment by paulirish** [comment]
Currently the maximum height is hardcoded in Chromium to that height. 
More information: https://groups.google.com/a/chromium.org/d/topic/headless-dev/DqaAEXyzvR0/discussion

**Comment by timhaines** [comment]
@paulirish It would be very nice if Chrome handled the stitching together of tall pages for you.  :)

**Comment by djones** [comment]
I have the same issue here. Here's another highly relevant thread https://bugs.chromium.org/p/chromium/issues/detail?id=638281

Two thoughts:

1) When the GPU is disabled, pure software rendering is used instead and therefore the max texture size is not relevant but unfortunately currently remains set. A flag could be added to Chrome, that when used in conjunction with `--disable-gpu`, would allow the max texture size to be far bigger. Something like `--max-texture-size=50000`. The current max texture size is hardcoded here:  https://cs.chromium.org/chromium/src/cc/resources/resource_provider.cc?l=344 and is very arbitrary.

2) If 1) is not feasible, a stitcher would have to be written to scroll, take snapshot, scroll etc. Perhaps something like this would be in the scope of `puppeteer`? Otherwise `page.screenshot({fullPage: true});` is a little misleading.

**Comment by paulirish** [comment]
I've put up the basic workaround for this bug here: #937

**Comment by guowenfh** [comment]
I testting to puppeteer1.7.0 node8.11.2.

screenshot images max height is 65000

```
page.screenshot({
   clip:{
    x:0,
    y:0,
    height:65000
  }
})
```

if height greater than 65000,  local image file is size to 0kb....

**Comment by adamchenwei** [comment]
lol @dgozman's solution works sooooo well! crazy fast. and it has been like... what two years? lol can't this be set to be some sort of default for puppeteer or a config to allow large pages render correctly in screenshot? or maybe add warning when this sort of large page render likely took place?

**Comment by schliflo** [comment]
@adamchenwei I can't find the solution you're talking about. Can you provide a link to it please? Thank you!

**Comment by yhafri** [comment]
> lol @dgozman's solution works sooooo well! crazy fast. and it has been like... what two years? lol can't this be set to be some sort of default for puppeteer or a config to allow large pages render correctly in screenshot? or maybe add warning when this sort of large page render likely took place?

@achenwei-chwy can you please point the solution?

**Comment by travisdowns** [comment]
Not sure if something has changed here, but I have been able to successfully take screenshots of > 50,000 pixels height using puppeteer 10.2.0 in headless mode.

**Comment by jrandolf-google** [comment]
Close due to staleness. Please reopen if the issue remains in the latest version.

**Comment by cooljacket** [comment]
> Close due to staleness. Please reopen if the issue remains in the latest version.

Hello jrandolf, are you sure the latest version dose fix the problem? I'm using the latest 15.3.1 version, while this problem hit me.

The viewport size is 9354 * 6622 with deviceScaleFactor 1.5, screenshot with 'fullPage: true', and incomplete in the right bottom side.

But when I just clip the right bottom part, it becomes complete picture.

So I guess that it maybe somewhere like an array has fix length, the screenshot data was cut down.

<img width="293" alt="image" src="https://user-images.githubusercontent.com/13379733/179979703-b3383225-2f03-4ad3-98a9-6d6b8ca950e5.png">

**Comment by matsuyama-k1** [comment]
I encountered this problem with the latest version of puppeteer(23.4.0).
So I solved this problem by taking screenshot separately then merge them.

https://github.com/matsuyama-k1/puppeteer-screenshot-larger-than-16384px/

![Screenshot 2024-09-24 at 2 47 23](https://github.com/user-attachments/assets/7b7e717f-14d2-4c94-969a-f3b637123429)

**Comment by matsuyama-k1** [comment]
I replaced the example repository to gist!
https://github.com/matsuyama-k1/puppeteer-screenshot-larger-than-16384px/

I didn't have time to adjust the repository so that it could be imported by cloning it, so I decided to publish a gist instead of the repository.

**Comment by jlahijani** [comment]
> > Close due to staleness. Please reopen if the issue remains in the latest version.
> 
> Hello jrandolf, are you sure the latest version dose fix the problem? I'm using the latest 15.3.1 version, while this problem hit me.
> 
> The viewport size is 9354 * 6622 with deviceScaleFactor 1.5, screenshot with 'fullPage: true', and incomplete in the right bottom side.
> 
> But when I just clip the right bottom part, it becomes complete picture.
> 
> So I guess that it maybe somewhere like an array has fix length, the screenshot data was cut down.
> 
> <img alt="image" width="293" src="https://user-images.githubusercontent.com/13379733/179979703-b3383225-2f03-4ad3-98a9-6d6b8ca950e5.png">

I'm not using Puppeteer but had this exact issue when using chrome headless directly.  I went insane trying to find the answer, but it's here:
https://github.com/puppeteer/puppeteer/issues/5530#issuecomment-605047166
